src/applications/coronavirus-screener/components/MultiQuestionForm.jsx

```diff
@@ -1,63 +1,123 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Element } from 'react-scroll';
 import FormQuestion from './FormQuestion';
-import { Element, scroller } from 'react-scroll';
 import FormResult from './FormResult';
+import recordEvent from 'platform/monitoring/record-event';
+import moment from 'moment';
 import _ from 'lodash/fp';
+import { getEnabledQuestions, checkFormStatus, scrollTo } from '../lib';
 
-// scoller usage based on https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/packages/formation-react/src/components/CollapsiblePanel/CollapsiblePanel.jsx
+export default function MultiQuestionForm({ questions, defaultOptions }) {
+  const [formState, setFormState] = useState({
+    status: 'incomplete',
+    startTime: null,
+    completed: false,
+  });
+
+  const [questionState, setQuestionState] = useState(questions);
 
-function scrollTo(name) {
-  scroller.scrollTo(
-    name,
-    window.VetsGov.scroll || {
-      duration: 500,
-      delay: 2,
-      smooth: true,
+  // updates formState based on questionState
+  // note: investigate https://reactjs.org/docs/hooks-reference.html#usereducer
+  useEffect(
+    () => {
+      let completed = formState.completed;
+      const newStatus = checkFormStatus(questionState);
+      if (formState.status !== newStatus) {
+        // record first completion of form
+        if (completed === false) {
+          recordEvent({
+            event: 'covid-screening-tool-result-displayed',
+            'screening-tool-result': formState.result,
```

Comment:
@jonwehausen we changed the `screening-tool-result` value to match our machine-readable result status string, does that seem reasonable? Since we might add more result options or change wording in the future.
For example, `'screening-tool-result': 'More screening needed'` becomes `'screening-tool-result': 'fail'

Reply:
hmm it looks like this is being used in the "Total and average time to completion" panel on https://analytics.google.com/analytics/web/?authuser=2#/dashboard/qHm9KfILQnmYWnUVDycH0w/a50123418w177519031p184624291/_u.date00=20200604&_u.date01=20200604/ so changing the value would affect those totals :thinking:

Reply:
Hi @mchelen-gov,

Thank you for thinking of us in this change. Changes look good to me -- understanding the `'screening-tool-result'` dataLayer variable will now populate with `'fail'`. Agree this is a better taxonomy moving forward. As far as the calculation goes, you are correct this will affect the view of that chart on dashboard. 

For context, we are currently calculating the "Average Screening Tool Completion Time" by taking Total Screening Time Time Spent / Number of Results returned. This then allows us to break the calculation down by passes & failures when coupling with the event action where we populate either `COVID Screening Tool - Result Displayed - Pass` or `COVID Screening Tool - Result Displayed - More screening needed` (as you currently see in the chart). 

As it stands, this change will add an additional row to that chart with `COVID Screening Tool - Result Displayed - fail` and it's broken-down calculation. Unfortunately, there's not a great way to merge the current `COVID Screening Tool - Result Displayed - More screening needed` syntax and the new `COVID Screening Tool - Result Displayed - fail` **within a single row of that current table**.

That said, we can add an additional metric box that consolidates both the old and soon-to-be new syntax. I've templatized that dashboard [here](https://analytics.google.com/analytics/web/template?uid=_mTuR6tNSyOjdKLbo5wRWw) if wishing to use. 

We are good to move forward with this change 👍

Reply:
@jonathanjnelson awesome, thank you for the input! modified the the failure case machine id to `more-screening` since that is more indicative of result